tests/gradients/core/test_hadamard_gradient.py

```diff
@@ -81,6 +81,22 @@ def cost9(x):
     return (qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1)))
 
 
+def cost10(x):
```

Comment:
What happens when the circuit returns a mix of `prod` and other measurements?

Reply:
Added `probs()` to another test case that mixes it with an expectation value. These tests are not set up to check mixed-shape returns, so I'd rather not introduce that additional complexity here :upside_down_face: